aws-logs-querydefinition/aws-logs-querydefinition.json

```diff
@@ -39,6 +39,10 @@
       "maxLength": 256
     }
   },
+  "required": [
+    "Name",
+    "QueryString"
+  ],
```

Comment:
This diff doesn’t seem to match the PR description?

Reply:
I'll update the description so its more clear. I have to find out more how CFN handles templates depending on what is specified in the required field. My integration tests before were working against a private resource type that had the changes in this PR, however I don't know if leaving out the required field entirely would make each prop needed in the template or optional - either way, this makes it more explicit. 

The reason why I'm unsure is that our other resource types have required fields that don't correspond directly to our actual API's required fields, but they are still working for customers AFAICT

Reply:
[`By default, the properties defined by the properties keyword are not required. However, one can provide a list of required properties using the required keyword.`](https://json-schema.org/understanding-json-schema/reference/object.html#required-properties)